### PR TITLE
🧪 DEV: Use -e install for tox runs

### PR DESCRIPTION
**Pull request by chrisjsewell**
Quick one: Improves startup time for test runs

### CODE REVIEW

**Comment by ramirezfranciscof** [comment]
So, taking advantage that this is a very succint modification, I thought I might use it to learn a bit more about the CI (I'm not familiar with the details of how it works). What do this 2 lines do exactly?

**Comment by chrisjsewell** [comment]
So normally tox "installs" the repository into the virtual environment before using it e.g. `pip install .`, with this option instead it uses `pip install -e .`. This means that it does not have to copy all the files somewhere, which can be slow, particularly for a package the size of aiida-core.

In general, I would definitely give tox a go. The big selling point, at least for me, is that it makes it easier to develop across multiple repositories, by standardising the test process and handling the setup and maintenance of test environments. 
For example already this month I have worked on 16 repos 😬 

image

In most of these I have tox setup so that I simply open the local repo folder, type `tox` and it runs the test suite in a virtual environment, which it ensures is up-to-date with the necessary dependencies.

**Comment by chrisjsewell** [comment]
Sneaking in an extra docker-compose file as well, just for use in local testing, if you want to use an isolated version of rabbitmq (it explains how to use it in the initial comments)

**Comment by chrisjsewell** [comment]
BTW, just messing around trying to get all the tests to pass locally on my macbook.
With https://github.com/aiidateam/aiida-core/pull/4379, initializing the rabbitmq docker, and setting up the ssh key as in `.github/workflows/setup.sh`

```
ssh-keygen -q -t rsa -b 4096 -N "" -f ~/.ssh/id_rsa
ssh-keygen -y -f ~/.ssh/id_rsa >> ~/.ssh/authorized_keys
ssh-keyscan -H localhost >> ~/.ssh/known_hosts
```

and enabling remote login:

image

I'm down to a few outstanding errors:

```
FAILED tests/cmdline/commands/test_daemon.py::TestVerdiDaemon::test_daemon_restart - AssertionError: SystemExit(<ExitCode.CRITICAL: 1>) is not None : Traceback (most recent call last):
FAILED tests/cmdline/commands/test_daemon.py::TestVerdiDaemon::test_daemon_start - AssertionError: 'daemon-error-not-running' != 'ok'
FAILED tests/cmdline/commands/test_daemon.py::TestVerdiDaemon::test_daemon_start_number - AssertionError: 'daemon-error-not-running' != 'ok'
FAILED tests/cmdline/commands/test_daemon.py::TestVerdiDaemon::test_daemon_start_number_config - AssertionError: 'daemon-error-not-running' != 'ok'
FAILED tests/sphinxext/test_workchain.py::test_workchain_build - AssertionError: assert 2 == 0
FAILED tests/sphinxext/test_workchain.py::test_broken_workchain_build - assert 'The broken workchain says hi!' in '\nException occurred:\n  File "/Users/chrisjsewell/Documents/GitHub/aiida_core_develop/aiida/mana...
```

**Comment by ramirezfranciscof** [comment]
So, just to check, are you still working on this or did you manage to fix the issues and are waiting for me to approve?

**Comment by chrisjsewell** [comment]
@ramirezfranciscof I resolved the conflict, so you can go ahead an approve

**Review by ramirezfranciscof — APPROVED** [review]
Thanks @chrisjsewell !

File: pyproject.toml [hunk 1]
```diff
@@ -68,7 +68,10 @@ legacy_tox_ini = """
 [tox]
 envlist = py37-django
 
-[testenv:py{36,37,38}-{django,sqla}]
+[testenv]
+usedevelop=True
+
+[testenv:py{35,36,37,38}-{django,sqla}]
```
Comment (sphuber) [inline pyproject.toml:74]:
```suggestion
[testenv:py{36,37,38}-{django,sqla}]
```

**Review by ramirezfranciscof — APPROVED** [review]
So, it is not very lengthy but here is the summary of what I understand this PR does that I would include in the commit. I'm not an expert of the CI so maybe @chrisjsewell and @sphuber can take a look and comment what other details should be explained in the message.

"CI: use -e install for tox + add docker-compose for isolated rabbitmq

* Using the `pip install -e .` option for tox runs Improves startup time for tests by preventing unnecessary copy of files.

* The docker-compose yml file allows to set up an isolated rabbitmq instance for local CI testing."

**Comment by sphuber** [comment]
Seems like a good description to me, just some minor formatting changes to respect 72 chars max:
```
CI: use `-e` install for tox + add docker-compose for isolated RabbitMQ

 * Using `pip install -e .` for tox runs improves startup time for tests
   by preventing unnecessary copy of files.
 * The docker-compose yml file allows to set up an isolated RabbitMQ
   instance for local CI testing.
```
Feel free to update the branch and squash merge with that message

**Pull request merged**